alea/models/blueice_extended_model.py

```diff
@@ -381,6 +413,40 @@ def _set_efficiency(self, source: dict, ll):
             )
         ll.add_shape_parameter(efficiency_name, anchors=(limits[0], limits[1]))
 
+    def store_real_data(self, file_name: str, real_data_list: list, metadata=None):
+        """Store real data in a file with toydata format.
+
+        Args:
+            file_name (str): Name of the file.
+            real_data_list (list): List of np.array of real data.
+
+        """
+        # check if real_data_list has the correct length
+        if len(real_data_list) != len(self.likelihood_names) - 1:
+            raise ValueError(
+                f"real_data_list must have length {len(self.likelihood_names) - 1} "
+                f"according to the number of likelihood terms in the model, "
+                f"but has length {len(real_data_list)}."
+            )
+        # check if the dtypes of the real data match the dtypes of the data generators
+        expected_dtypes = [np.dtype(gen.dtype) for gen in self.data_generators]
+        if any([r_d.dtype != e_d for r_d, e_d in zip(real_data_list, expected_dtypes)]):
+            raise ValueError(
+                "The dtypes of the real data do not match the dtypes of the data generators."
+            )
+        # mimic the generate_values
+        _generate_values = self.parameters()
+        generate_values = dict_to_structured_array(_generate_values)
+        # mimic the ancillary_measurements
+        ancillary_keys = self.parameters.with_uncertainty.names
+        _ancillary_measurements = {k: v for k, v in _generate_values.items() if k in ancillary_keys}
+        ancillary_measurements = dict_to_structured_array(_ancillary_measurements)
```

Comment:
What we do here (which is fair I think) is to set the ancillary measurements to their nominal values. However, I'd write it so that this becomes clearer
```suggestion
        # set ancillary_measurements to nominal values
        ancillary_keys = self.parameters.with_uncertainty.names
        _ancillary_measurements = {p.name: p.nominal_value for p in self.parameters if p.name in ancillary_keys}
        ancillary_measurements = dict_to_structured_array(_ancillary_measurements)
```

An even shorter version would be:
```suggestion
         # set ancillary_measurements to nominal values
        _ancillary_measurements = parameters.with_uncertainty.nominal_values
        ancillary_measurements = dict_to_structured_array(_ancillary_measurements)
```
But it's maybe not as readable.

Reply:
I applied the 2nd suggestion.